src/web/rustdoc.rs

```diff
@@ -301,22 +289,37 @@ pub fn rustdoc_html_server_handler(req: &mut Request) -> IronResult<Response> {
     let latest_version = crate_details.latest_version().to_owned();
     let is_latest_version = latest_version == version;
 
-    let path = if !is_latest_version {
-        req_path[2] = &latest_version;
-        path_for_version(&req_path, &crate_details.target_name, &conn)
+    let path_in_latest = if !is_latest_version {
+        let mut latest_path = req_path.clone();
+        latest_path[2] = &latest_version;
+        path_for_version(&latest_path, &crate_details.target_name, &conn)
     } else {
         Default::default()
     };
 
+    // The path within this crate version's rustdoc output
+    let inner_path = {
+        let mut inner_path = req_path.clone();
+        // Drop the `rustdoc/:crate/:version[/:platform]` prefix
+        inner_path.drain(..3);
+        // This could be a top-level file like rustdoc/:crate/:version/settings.html, hopefully
+        // rustdoc won't ever output files that are under directories other than the `target_name`
+        if inner_path.len() > 1 && inner_path[0] != crate_details.target_name {
```

Comment:
Can you change this to instead check against a whitelist of the targets, instead of assuming all directories are targets? @GuillaumeGomez said that rustdoc may occasionally output files that aren't under `/target_name`. You should be able to get the targets from `crate_details`.